frontends/PyCDE/src/pycde/value.py

```diff
@@ -0,0 +1,72 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from __future__ import annotations
+
+from .support import get_user_loc
+
+import circt.support as support
+from circt.dialects import hw, seq
+
+import mlir.ir as ir
+
+
+class Value:
+  @staticmethod
+  def get(value, type=None):
+    from .pycde_types import PyCDEType
+    value = support.get_value(value)
+    if type is None:
+      type = value.type
+    type = PyCDEType(type)
+
+    if isinstance(type, hw.ArrayType):
+      return ListValue()._init(value, type)
+    if isinstance(type, hw.StructType):
+      return StructValue()._init(value, type)
+    return Value()._init(value, type)
+
+  def _init(self, value, type):
```

Comment:
any reason this isn't `def __init__` for a standard constructor? seems like that could work and would be a little more familiar for the "init" method.

Reply:
I don't want users calling it since if the call it with `Value(listValue, listType)`, it won't work properly. What a user means to call 99% of the time is `Value.get(listValue)`. I could create a separate constructor for each subclass, but that's more code... but it's probably clearer. I hate duplicated code, but... I'll make the change. *grumble* protected access level *grumble*.